alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertIndices.kt

```diff
@@ -161,45 +206,74 @@ class AlertIndices(
+                .scheduleWithFixedDelay({ rolloverAndDeleteFindingHistoryIndices() }, findingHistoryRolloverPeriod, executorName())
         }
     }
 
-    fun isInitialized(): Boolean {
-        return alertIndexInitialized && historyIndexInitialized
+    fun isAlertInitialized(): Boolean {
+        return alertIndexInitialized && alertHistoryIndexInitialized
     }
 
-    fun isHistoryEnabled(): Boolean = historyEnabled
+    fun isAlertHistoryEnabled(): Boolean = alertHistoryEnabled
+
+    fun isFindingHistoryEnabled(): Boolean = findingHistoryEnabled
 
     suspend fun createOrUpdateAlertIndex() {
         if (!alertIndexInitialized) {
-            alertIndexInitialized = createIndex(ALERT_INDEX)
+            alertIndexInitialized = createIndex(ALERT_INDEX, alertMapping())
             if (alertIndexInitialized) IndexUtils.alertIndexUpdated()
         } else {
-            if (!IndexUtils.alertIndexUpdated) updateIndexMapping(ALERT_INDEX)
+            if (!IndexUtils.alertIndexUpdated) updateIndexMapping(ALERT_INDEX, alertMapping())
         }
         alertIndexInitialized
     }
 
-    suspend fun createOrUpdateInitialHistoryIndex() {
-        if (!historyIndexInitialized) {
-            historyIndexInitialized = createIndex(HISTORY_INDEX_PATTERN, HISTORY_WRITE_INDEX)
-            if (historyIndexInitialized)
-                IndexUtils.lastUpdatedHistoryIndex = IndexUtils.getIndexNameWithAlias(clusterService.state(), HISTORY_WRITE_INDEX)
+    suspend fun createOrUpdateInitialAlertHistoryIndex() {
+        if (!alertHistoryIndexInitialized) {
+            alertHistoryIndexInitialized = createIndex(ALERT_HISTORY_INDEX_PATTERN, alertMapping(), ALERT_HISTORY_WRITE_INDEX)
+            if (alertHistoryIndexInitialized)
+                IndexUtils.lastUpdatedAlertHistoryIndex = IndexUtils.getIndexNameWithAlias(
+                    clusterService.state(),
+                    ALERT_HISTORY_WRITE_INDEX
+                )
         } else {
-            updateIndexMapping(HISTORY_WRITE_INDEX, true)
+            updateIndexMapping(ALERT_HISTORY_WRITE_INDEX, alertMapping(), true)
         }
-        historyIndexInitialized
+        alertHistoryIndexInitialized
     }
 
-    private suspend fun createIndex(index: String, alias: String? = null): Boolean {
+    suspend fun createOrUpdateInitialFindingHistoryIndex() {
+        if (!findingHistoryIndexInitialized) {
+            findingHistoryIndexInitialized = createIndex(FINDING_HISTORY_INDEX_PATTERN, findingMapping(), FINDING_HISTORY_WRITE_INDEX)
+            if (findingHistoryIndexInitialized) {
+                IndexUtils.lastUpdatedFindingHistoryIndex = IndexUtils.getIndexNameWithAlias(
+                    clusterService.state(),
+                    FINDING_HISTORY_WRITE_INDEX
+                )
+            }
+        } else {
+            updateIndexMapping(FINDING_HISTORY_WRITE_INDEX, findingMapping(), true)
```

Comment:
Do we need to update mapping for every monitor run action? Or it is more of a defensive approach in the absence of any better option.

Reply:
This is to ensure the findings schema is up to date when there are cluster upgrades and the finding schema got updated

Reply:
Yeah, but this logic will end up being invoked to update the mapping with every run .